src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -3649,6 +3653,23 @@ public List<GHRepositoryTrafficTopReferralSources> getTopReferralSources() throw
                 .fetch(GHRepositoryTrafficTopReferralSources[].class));
     }
 
+    /**
+     * Get all active rules that apply to the specified branch
+     * (https://docs.github.com/en/rest/repos/rules?apiVersion=2022-11-28#get-rules-for-a-branch).
+     *
+     * @param branch
+     *            the branch
+     * @return the rules for branch
+     * @throws IOException
+     *             the io exception
+     */
+    public List<GHRepositoryRule> getRulesForBranch(String branch) throws IOException {
+        return Arrays.asList(root().createRequest()
+                .method("GET")
+                .withUrlPath(getApiTailUrl("/rules/branches/" + branch))
+                .fetch(GHRepositoryRule[].class));
```

Comment:
Are these repository rules or branch rules?  The rulesets are definitely repository rulesets, but I think these are the resulting rules for a specific branch.   For example you might have two different repository rulesets that apply to branch. I think the result of the call would be the union of those two rulesets, but you're going to new bunch of tests to show that.

Reply:
The endpoint provides the union of applicable org and repo rulesets for a branch. What I am not sure about is, if "normal" repository rules will be taken into account as well.

Reply:
In the GraphQL Schema they are defined as `type RepositoryRule implements Node {` so I guess this name is correct.